Model/Adminhtml/Config/Insurance/ConfigurationSave.php

```diff
@@ -22,46 +23,56 @@ class ConfigurationSave extends Value
      * @var Logger
      */
     private $logger;
-    private $productRepository;
+    /**
+     * @var InsuranceProductHelper
+     */
     private $insuranceProductHelper;
+    /**
+     * @var QuoteHelper
+     */
+    private $quoteHelper;
 
     public function __construct(
         Context                    $context,
         Registry                   $registry,
         ScopeConfigInterface       $config,
         TypeListInterface          $cacheTypeList,
         Logger                     $logger,
-        ProductRepositoryInterface $productRepository,
         InsuranceProductHelper     $insuranceProductHelper,
+        QuoteHelper                $quoteHelper,
         AbstractResource           $resource = null,
         AbstractDb                 $resourceCollection = null,
         array                      $data = []
     ) {
         parent::__construct($context, $registry, $config, $cacheTypeList, $resource, $resourceCollection, $data);
         $this->logger = $logger;
-        $this->productRepository = $productRepository;
         $this->insuranceProductHelper = $insuranceProductHelper;
+        $this->quoteHelper = $quoteHelper;
     }
 
     /**
-     * @throws AlmaInsuranceProductException
      * @return $this
+     * @throws AlmaInsuranceProductException
      */
     public function beforeSave(): ConfigurationSave
     {
         $arrayValue = json_decode($this->getValue(), true);
+        if (!$this->isValueChanged()) {
+            return $this;
+        }
         if (
-            !$this->isValueChanged() ||
             !isset($arrayValue['isInsuranceActivated']) ||
             $arrayValue['isInsuranceActivated'] === false
         ) {
+            $this->insuranceProductHelper->disableInsuranceProductIfExist();
+            $this->quoteHelper->deleteInsuranceDataFromQuoteItemForNotConvertedQuote();
             return $this;
         }
 
         try {
-            $this->productRepository->get(InsuranceHelper::ALMA_INSURANCE_SKU);
+            $this->insuranceProductHelper->enableInsuranceProductIfExist();
```

Comment:
You always enable the product if the insurance is activated ?
Why don't add a condition like the disable case ?
If the merchant disable manually the insurance product by chose ?

Reply:
Test is already enabled directly in service.
If merchant disable manually, impossible to add insurance in quote. Bug, need to check if product is enable for widget display